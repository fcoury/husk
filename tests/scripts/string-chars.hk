// Test string.chars() method
// Tests Unicode handling, empty strings, and various character types

// Basic ASCII string
let hello = "hello";
let hello_chars = hello.chars();
println!(hello_chars.len());
println!(hello_chars[0]);
println!(hello_chars[4]);

// Empty string
let empty = "";
let empty_chars = empty.chars();
println!(empty_chars.len());

// Unicode string with accented characters
let cafe = "cafÃ©";
let cafe_chars = cafe.chars();
println!(cafe_chars.len());
println!(cafe_chars[3]);

// Emoji (tests surrogate pair handling)
let emoji = "Hello ðŸ‘‹ World";
let emoji_chars = emoji.chars();
println!(emoji_chars.len());
println!(emoji_chars[6]);

// Multiple emojis
let emojis = "ðŸŽ‰ðŸŒŸâœ¨";
let emojis_chars = emojis.chars();
println!(emojis_chars.len());
println!(emojis_chars[0]);
println!(emojis_chars[2]);

// Mixed scripts (Latin + Chinese)
let mixed = "Helloä¸–ç•Œ";
let mixed_chars = mixed.chars();
println!(mixed_chars.len());
println!(mixed_chars[5]);
println!(mixed_chars[6]);

// Test that each element is a single-character string
let test = "testðŸ‘‹";
let test_chars = test.chars();
println!(test_chars.len());
println!(test_chars[4]);

// String with newlines and tabs
let special = "a\tb\nc";
let special_chars = special.chars();
println!(special_chars.len());
println!(special_chars[1]);
println!(special_chars[3]);