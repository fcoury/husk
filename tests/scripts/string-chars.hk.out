5
h
o
0
4
é
13
👋
3
🎉
✨
7
世
界
5
👋
5
	


